### PR TITLE
[LRN-396]: Add Ability to Upload Attachments to Course Syllabus field in Manual Verification Form

### PR DESCRIPTION
**Pull request by justinmckenzie**
Ticket: https://jira.mongodb.org/browse/LRN-396

This PR edits the "Course Syllabus" field on the existing Manual Verification form to accept either an attachment or a url to a document publicly hosted elsewhere. Since AirTable does not support including attachments in their API to then be hosted on their side, we are instead adding those attachments to a dedicated Google Drive folder and taking the direct url to the file and uploading it to the "Course Syllabus" field in AirTable.

### CODE REVIEW

File: src/pages/api/registration.ts [hunk 1]
```diff
@@ -36,31 +46,54 @@ const registrationHandler = async (
...
-  const message = 'Success';
+  const registrationToUpload = getDBRegistration(
+    fields as unknown as Registration
+  );
 
+  // TODO: should really have error handling if form doesnt submit at least to Airtable?
   if (
-    process.env['APP_ENV'] === 'production' ||
-    process.env['APP_ENV'] === 'staging'
+    process.env.APP_ENV === 'staging' ||
```
Comment (justinmckenzie) [inline src/pages/api/registration.ts:83]: leaving this for now for UAT

File: src/components/form/custom-fields/course-syllabus.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  function onUploadDocClick() {
+    setField('fileUpload');
+  }
+
+  function onEnterUrlClick() {
+    setField('webUrl');
+  }
+
+  return (
+    <div>
```
Comment (gfunk1230) [inline src/components/form/custom-fields/course-syllabus.tsx:22]: Do we need this `div` here, or could we use a react fragment `<>`?

File: src/components/form/custom-fields/course-syllabus.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      {field === 'fileUpload' && (
+        <label htmlFor="syllabus-upload">
+          <input
+            type="file"
+            accept=".pdf, .doc, .docx"
+            id="file-syllabus-upload"
+            name="file-syllabus-upload"
+            data-testid="file-syllabus-upload"
+            sx={styles.SyllabusFileUpload}
+            onChange={({
```
Comment (gfunk1230) [inline src/components/form/custom-fields/course-syllabus.tsx:44]: Could we hoist this up in a function just to make it cleaner?

File: src/components/form/custom-fields/course-syllabus.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          }}
+        >
+          <input
+            type="text"
+            placeholder="Enter Web URL"
+            id="url-syllabus-upload"
+            name="url-syllabus-upload"
+            data-testid="url-syllabus-upload"
+            onFocus={() => setInputFocus(true)}
+            onBlur={({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
```
Comment (gfunk1230) [inline src/components/form/custom-fields/course-syllabus.tsx:68]: Could we hoist this up in a function just to make it cleaner?

File: package.json [hunk 5]
```diff
@@ -26,8 +26,11 @@
     "@mdb/flora": "1.3.5",
     "@mdx-js/react": "2.2.1",
     "@sentry/nextjs": "7.36.0",
+    "@types/formidable": "2.0.5",
     "airtable": "0.11.6",
     "axios": "1.3.2",
+    "formidable": "2.1.1",
```
Comment (gfunk1230) [inline package.json:32]: no biggie, but I just wanted to ask why we are not locking into specific versions for our dependencies.

**Review by gfunk1230 — APPROVED** [review]
I left minor comments but LGTM. I would just have @mcolella14 take a look as well.

File: package.json (hunk 5 above)
Comment (justinmckenzie) [reply package.json:32]: this is locked into a specific version?

File: src/components/form/custom-fields/course-syllabus.tsx (hunk 2 above)
Comment (justinmckenzie) [reply src/components/form/custom-fields/course-syllabus.tsx:22]: yeah need this block element for layout of children

**Comment by justinmckenzie** [comment]
merging to test staging before UAT meeting, will address any outstanding code review comments in additional PR

**Pull request merged**